lib/poparser.js

```diff
@@ -6,25 +6,50 @@ const util = require('util');
 /**
  * Parses a PO object into translation table
  *
- * @param {Buffer|String} buffer PO object
- * @param {String} [defaultCharset] Default charset to use
- * @return {Object} Translation object
+ * @param buffer PO object
+ * @param defaultCharset Default charset to use
+ * @param validation Enables validation of parsed content, if set to true
+ * @type {{
+ *    (buffer: string | Buffer): object;
+ *    (buffer: string | Buffer, defaultCharset: string): object;
+ *    (buffer: string | Buffer, defaultCharset: string, validation: boolean): object;
+ *    (buffer: string | Buffer, validation: boolean): object;
+ * }}
  */
-module.exports.parse = function (buffer, defaultCharset) {
-  const parser = new Parser(buffer, defaultCharset);
+module.exports.parse = function (buffer, defaultCharset, validation) {
+  const args = sharedFuncs.unshiftOptionalArgs(
```

Comment:
At first sight, I'd prefer to use a method signature like this:
`parse = function (buffer, options) {`
With `options` being an object with optional properties like `defaultCharset` and `validation`. To prevent breaking current usage, i'd say a `typeof options === 'string'` check (meaning `options` is actually`defaultCharset`) would be enough.
In my opinion this facilitates additional options in the future while not needing the 'magic' `unshiftOptionalArgs` provides. But please tell me if I'm missing something with this proposal.

Reply:
https://github.com/smhg/gettext-parser/pull/76#discussion_r1134071829